logbus/ship/ship.go

```diff
@@ -53,7 +52,7 @@ func (l *LogShipper) Write(delta *pb.Delta) {
 // Start the log streaming process and begin writing logs to the server.
 func (l *LogShipper) Start(ctx context.Context) {
 	go func() {
-		ctx, l.cancel = context.WithCancel(ctx)
+		ctx, l.cancel = context.WithCancel(context.Background())
```

Comment:
jsut curious is this because ctx is already canceled?

Reply:
When the parent context was canceled, the child was also canceled. This killed the log streaming process and led to a deadlock.